src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -1855,7 +1884,7 @@ internal int ANSICodePage
             {
                 if (_iDefaultAnsiCodePage == undef && !GlobalizationMode.Invariant)
                 {
-                    _iDefaultAnsiCodePage = GlobalizationMode.UseNls ? NlsGetAnsiCodePage(_sRealName!) : IcuGetAnsiCodePage(_sRealName!);
+                    _iDefaultAnsiCodePage = GlobalizationMode.UseNls ? NlsGetLocaleInfo(LocaleNumberData.AnsiCodePage) : IcuGetAnsiCodePage(_sRealName!);
```

Comment:
>NlsGetLocaleInfo [](start = 71, length = 16)

for codepages, we can always read them from NLS if we are running on Windows OS. users cannot override these and ICU doesn't provide them. We carry our own data to use when running on Linux/OSX. this comment apply to the 4 codepages.

Reply:
Ok, will do that.